mailpoet/assets/js/src/segments/segments.tsx

```diff
@@ -42,19 +42,19 @@ function Tabs() {
 
 Tabs.displayName = 'SegmentTabs';
 
-function App() {
+function App(): JSX.Element {
   return (
     <GlobalContext.Provider value={useGlobalContextValue(window)}>
       <HashRouter>
         <Notices />
         <Switch>
           <Route exact path="/" render={() => <Redirect to="/lists" />} />
-          <Route path="/new" render={withBoundary(SegmentForm)} />
-          <Route path="/edit/:id" render={withBoundary(SegmentForm)} />
-          <Route path="/new-segment" render={withBoundary(Editor)} />
-          <Route path="/edit-segment/:id" render={withBoundary(Editor)} />
-          <Route path="/segments/(.*)?" render={withBoundary(Tabs)} />
-          <Route path="/lists/(.*)?" render={withBoundary(Tabs)} />
+          <Route path="/new" component={withBoundary(SegmentForm)} />
+          <Route path="/edit/:id" component={withBoundary(SegmentForm)} />
+          <Route path="/new-segment" component={withBoundary(Editor)} />
+          <Route path="/edit-segment/:id" component={withBoundary(Editor)} />
+          <Route path="/segments/(.*)?" component={withBoundary(Tabs)} />
+          <Route path="/lists/(.*)?" component={withBoundary(Tabs)} />
```

Comment:
Was the change from `render` to `component` necessary? With the `component` prop, a new component is instantiated on every render, which may not be ideal. From https://v5.reactrouter.com/web/api/Route/component:

> When you use component (instead of render or children, below) the router uses [React.createElement](https://facebook.github.io/react/docs/react-api.html#createelement) to create a new [React element](https://facebook.github.io/react/docs/rendering-elements.html) from the given component. That means if you provide an inline function to the component prop, you would create a new component every render. This results in the existing component unmounting and the new component mounting instead of just updating the existing component.

Reply:
Without this change, I get the following error after converting the file to TSX:

```
TS2769: No overload matches this call.
Overload 1 of 2, '(props: (RouteProps<string, { [x: string]: string; }> & OmitNative<{}, keyof RouteProps<string, { [x: string]: string; }>>) | 
Readonly<RouteProps<string, { ...; }> & OmitNative<...>>): Route<...>', gave the following error.     Type 'ComponentType<SegmentFormPropType>' is not assignable to type '(props: RouteComponentProps<{ [x: string]: string; }, StaticContext, unknown>) => ReactNode'.
Type 'ComponentClass<SegmentFormPropType, any>' is not assignable to type '(props: RouteComponentProps<{ [x: string]: string; }, StaticContext, unknown>) => ReactNode'.
Type 'ComponentClass<SegmentFormPropType, any>' provides no match for the signature '(props: RouteComponentProps<{ [x: string]: string; }, StaticContext, unknown>): ReactNode'.
Overload 2 of 2, '(props: RouteProps<string, { [x: string]: string; }> & OmitNative<{}, keyof RouteProps<string, { [x: string]: string; }>>, context: any): Route<...>', gave the following error.
Type 'ComponentType<SegmentFormPropType>' is not assignable to type '(props: RouteComponentProps<{ [x: string]: string; }, StaticContext, unknown>) => ReactNode'.
Type 'ComponentClass<SegmentFormPropType, any>' is not assignable to type '(props: RouteComponentProps<{ [x: string]: string; }, StaticContext, unknown>) => ReactNode'.

index.d.ts(92, 5): The expected type comes from property 'render' which is declared here on type 'IntrinsicAttributes & IntrinsicClassAttributes<Route<{}, string>> & Readonly<RouteProps<string, { [x: string]: string; }> & OmitNative<...>>'
index.d.ts(92, 5): The expected type comes from property 'render' which is declared here on type 'IntrinsicAttributes & IntrinsicClassAttributes<Route<{}, string>> & Readonly<RouteProps<string, { [x: string]: string; }> & OmitNative<...>>'
```

I'm not sure if there is a better way to fix this error. I followed what I found in https://github.com/mailpoet/mailpoet/blob/91e333fa6cfffd9a16b3482b55ace103608c8152/mailpoet/assets/js/src/wizard/wizard.tsx#L23.

Reply:
@rodrigoprimo Ahh, `render` expects a function, so I guess what you have now is actually correct, as long as the HOC (`withBoundary`) behaves well. So seems fine 👍